File: app/src/main/java/xyz/jtmiles/beastforgw2/services/CharactersService.java
package xyz.jtmiles.beastforgw2.services;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Path;
import retrofit2.http.Query;
import xyz.jtmiles.beastforgw2.models.Character;

/**
 * Created by JT on 5/15/2016.
 */
public interface CharactersService {
    @GET("characters")
    Call<List<String>> getCharacterNames(@Header("Authorization") String authorization);

    @GET("characters/{name}")
    Call<Character> getCharacterByName(@Path("name") String characterName, @Header("Authorization") String authorization);

    @GET("characters")
    Call<List<Character>> getCharactersByPage(@Query("page") int page, @Header("Authorization") String authorization);

}

File: app/src/main/java/xyz/jtmiles/beastforgw2/services/CharactersService.kt
package xyz.jtmiles.beastforgw2.services

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.Path
import retrofit2.http.Query
import xyz.jtmiles.beastforgw2.models.Character

interface CharactersService {
    @GET("characters")
    fun getCharacterNames(@Header("Authorization") authorization: String): Call<List<String>>

    @GET("characters/{name}")
    fun getCharacterByName(@Path("name") characterName: String, @Header("Authorization") authorization: String): Call<Character>

    @GET("characters")
    fun getCharactersByPage(@Query("page") page: Int, @Header("Authorization") authorization: String): Call<List<Character>>

}
